eth/get_proof.go

```diff
@@ -33,17 +36,41 @@ func (n *proofList) Delete(key []byte) error {
 	panic("not supported")
 }
 
-func NewProofProvider() *ProofProvider {
-	return &ProofProvider{}
+func NewProofProvider(db state.Database, blockStore store.BlockStoreReader) *ProofProvider {
+	return &ProofProvider{
+		database:   db,
+		blockStore: blockStore,
+	}
 }
 
 // getState returns the state.StateBD and Header of block at the given number.
 // If the passed number is nil, it returns the the db and header of the latest block.
 //
-// TODO: replace this with a real implementation based on `ProofProvider` being supplied
+// WIP: replace this with a real implementation based on `ProofProvider` being supplied
```

Comment:
should this `WIP` comment be removed? Or is to show that the impl may change once the unit test is updated to be more than a stub?

Reply:
Yup. 9c0fe5a